sentry_sdk/scope.py

```diff
@@ -1251,3 +1415,69 @@ def __repr__(self):
             hex(id(self)),
             self._name,
         )
+
+
+def _with_new_scope():
+    # type: () -> Generator[Scope, None, None]
+
+    current_scope = Scope.get_current_scope()
+    forked_scope = current_scope.fork()
```

Comment:
I do not understand why we use BOTH `copy_context` and fork. It seems both serve the same purpose?

Reply:
I checked with Francesco and indeed both is not necessary. Good catch!